sans/direct-beam.ipynb

```diff
@@ -0,0 +1,529 @@
+    "                                         direct_beam=direct_beam_by_pixel,\n",
+    "                                         direct_beam_transmission=background_trans, # note: background_trans\n",
+    "                                         masks=masks,\n",
+    "                                         wavelength_bands=wavelength_bands)\n",
+    "    background_q_lambda = to_q_by_wavelength(data=background,\n",
+    "                                             transmission=background_trans,\n",
+    "                                             direct_beam=direct_beam_by_pixel,\n",
+    "                                             direct_beam_transmission=background_trans, # note: same as transmission\n",
+    "                                             masks=masks,\n",
+    "                                             wavelength_bands=wavelength_bands)\n",
+    "\n",
+    "    # reduced by wavelength band\n",
+    "    reduced_by_wavelength = normalize_and_subtract(sample_q_lambda, background_q_lambda)\n",
+    "    \n",
+    "    # sum wavelength bands to reduce for full range\n",
+    "    sample_q1d = sc.sum(sample_q_lambda, 'wavelength')\n",
+    "    background_q1d = sc.sum(background_q_lambda, 'wavelength')\n",
+    "    reduced = normalize_and_subtract(sample_q1d, background_q1d)\n",
+    "    \n",
+    "    params, diff = fit_gauss_coil(contrib.select_bins(reduced, 'Q', qlongW['bound', 0], qshortW['bound', 1]),\n",
+    "                                  flat_background=flat_background)\n",
+    "    scale = I0_expected / params['parameter', 0].data\n",
+    "    reduced_by_wavelength *= scale\n",
+    "    reduced *= scale\n",
+    "    direct_beam = direct_beam / scale\n",
+    "    ratio = (reduced_by_wavelength - flat_background) / (reduced - flat_background)\n",
+    "    ratio.masks['Q-range-mask'] = q_range_mask(wavelength_bands)\n",
+    "    norm = 1.0*sc.units.one + damp*(sc.mean(ratio, 'Q') - 1.0*sc.units.one)\n",
+    "    # TODO original uses scale for interpolation, but is this necessary if we don't deal with histogram?\n",
+    "    # d_w = wavelength['wavelength',1:]-wavelength['wavelength',:-1]\n",
+    "    # scale = 1.0*sc.units.angstrom + damp*(norm*d_w-1.0*sc.units.angstrom)\n",
+    "    splined = interpolate_cubic_spline(norm, direct_beam.coords['wavelength'], 'wavelength')\n",
+    "    direct_beam *= splined\n",
+    "    \n",
+    "    # store some information for inspecting intermediate terms\n",
+    "    direct_beam.attrs['fit'] = sc.Variable(value=diff)\n",
+    "    direct_beam.attrs['knots'] = sc.Variable(value=norm)\n",
+    "    direct_beam.attrs['spline'] = sc.Variable(value=splined)\n",
+    "    \n",
+    "    return direct_beam"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dataconfig # run make_config.py to create this\n",
```

Comment:
maybe catch import error and provide this information. Judith got stuck here.

Reply:
:+1: